discord/audit_logs.py

```diff
@@ -188,15 +188,26 @@ def _transform(entry: AuditLogEntry, data: Optional[str]) -> Optional[Asset]:
 def _transform_automod_trigger_metadata(
     entry: AuditLogEntry, data: AutoModerationTriggerMetadata
 ) -> Optional[AutoModTrigger]:
+
+    if isinstance(entry.target, AutoModRule):
+        # type of trigger cannot be changed
+        # try to get trigger type from the resolved rule
+        try:
+            return AutoModTrigger.from_data(type=entry.target.trigger.type.value, data=data)
+        except Exception:
+            pass
```

Comment:
What is this doing?

Reply:
The trigger type can't be changed, so if the AutoModRule is available the trigger type should be the same instead of having to check which keys are in the data.

Found trying to edit with invalid trigger metadata such as `{'bogus': 'value'}` was accepted and the data from audit log could be `{}`, not sure if fetched versions behave the same. This can still get the trigger type and returns a trigger vs None.

Reply:
Probably should document that better in the comments